File: lecture015/dialPad.java
import java.util.Scanner;
import java.io.StringReader;
import java.util.ArrayList;

public class dialPad {
    public static Scanner scn = new Scanner(System.in);

    public static void main(String[] args) {
        solve();
    }

    public static void solve() {
        dialPad_("235", "");
    }


    public static int dialPad_(String ques, String ans) {
        if (ques.length() == 0) {
            // ArrayList<String> base = new ArrayList<>();
            // base.add("");
            // return base;
            System.out.println(ans);
            return 1;
        }
        char codes[] = { "+", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wx", "yz", "*", "#" };

        int count = 0;
        int ch = ques.get(0);
        String roq = ques.subString(1);
        String code = codes.get(ch - '0');
        for (int i = 0; i < code.length(); i++) {
            count += dialPad(roq, ans + code[i]);
        }
        return count;
    }
}

File: lecture015/dialPad.cpp
#include <iostream>
#include <string>
using namespace std;

string codes[] = {"+", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wx", "yz", "*", "#"};

int dialPad(string ques, string ans)
{
    if (ques.length() == 0)
    {
        cout << ans << " ";
        return 1;
    }
    int count = 0;
    char ch = ques[0];
    string roq = ques.substr(1);
    string code = codes[ch - '0'];

    for (int i = 0; i < code.length(); i++)
    {
        count += dialPad(roq, ans + code[i]);
    }
    return count;
}
void solve()
{
    cout << endl << dialPad("235", "");
}
int main(int args, char **argv)
{
    solve();
    return 0;
}
